src/components/badges/ConfettiIfNeeded.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useRef } from 'preact/hooks'
+import { useSnapshot } from 'valtio'
+import classnames, {
+  height,
+  inset,
+  pointerEvents,
+  position,
+  width,
+  zIndex,
+} from 'classnames/tailwind'
+import confetti from 'canvas-confetti'
+import walletStore from 'stores/WalletStore'
+
+const confettiCanvas = classnames(
+  position('absolute'),
+  inset('lg:-left-32', 'left-0'),
+  width('lg:w-full-125', 'w-full'),
+  height('h-full'),
+  zIndex('z-50'),
+  pointerEvents('pointer-events-none')
+)
+
+export default function () {
+  const { firstBadge } = useSnapshot(walletStore)
+  const canvasRef = useRef<HTMLCanvasElement>(null)
+
+  useEffect(() => {
+    if (!canvasRef.current) return
+    const mintConfetti = confetti.create(canvasRef.current, { resize: true })
+
+    void mintConfetti({
+      spread: 80,
+      decay: 0.8,
+      particleCount: 150,
+      colors: ['#fed823', '#ff7bed', '#15a1fc', '#01feb6'],
+    })
+  }, [canvasRef, firstBadge])
+
+  const shouldDisplay =
+    firstBadge.minted && !firstBadge.notified && !firstBadge.twitted
+
+  return shouldDisplay ? (
+    <canvas
```

Comment:
If we don't use our own canvas, it will create it's own in the center of the screen